src/nspawn/nspawn.c

```diff
@@ -2317,14 +2317,16 @@ static int inner_child(
                 memcpy_safe(a + 1, arg_parameters, m * sizeof(char*));
                 a[1 + m] = NULL;
 
-                exec_target = a[0] = (char*) "/usr/lib/systemd/systemd";
+                a[0] = (char*) "/usr/lib/systemd/systemd";
                 execve(a[0], a, env_use);
 
-                exec_target = a[0] = (char*) "/lib/systemd/systemd";
+                a[0] = (char*) "/lib/systemd/systemd";
                 execve(a[0], a, env_use);
 
-                exec_target = a[0] = (char*) "/sbin/init";
+                a[0] = (char*) "/sbin/init";
                 execve(a[0], a, env_use);
+
+                exec_target = "/usr/lib/systemd/systemd + /lib/systemd/systemd + /sbin/init";
```

Comment:
Maybe "/usr/lib/systemd/systemd, /lib/systemd/systemd, /sbin/init"? A bit shorter and easier to understand.